### PR TITLE
#220 BkBasic and BkParallel are not tested

### PR DESCRIPTION
**Pull request by dmzaytsev**
For #220 

created unit test for BkBasic

### CODE REVIEW

**Comment by davvd** [comment]
@dmzaytsev Thanks, let me find someone who can review this pull request

**Comment by davvd** [comment]
@pinaf it's yours, please review

File: src/test/java/org/takes/http/BkBasicTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * Test case for {@link BkBasic}.
+ *
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 0.15.2
+ */
+public final class BkBasicTest {
+    /**
+     * BkBasic can handle socket data.
+     * @throws java.io.IOException If some problem inside
```
Comment (pinaf) [inline src/test/java/org/takes/http/BkBasicTest.java]: @dmzaytsev no need for `java.io`, just `IOException` is fine.

File: src/test/java/org/takes/http/BkParallelTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ */
+package org.takes.http;
+
+/**
+ * Test case for {@link BkParallel}.
+ *
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 0.15.2
+ * @todo #220:30min/DEV This unit test not implemented but has to be.
```
Comment (pinaf) [inline src/test/java/org/takes/http/BkParallelTest.java]: @dmzaytsev let's make the description more verbose (at least 20 words).

**Comment by pinaf** [comment]
@dmzaytsev looks good - just 2 comments.

**Comment by dmzaytsev** [comment]
@pinaf I fixed javadocs
take a look please

**Comment by pinaf** [comment]
@rultor merge
@dmzaytsev thanks!

File: src/test/java/org/takes/http/BkParallelTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ *  You should mock Executor service and verify execute() calls.
```
Comment (yegor256) [inline src/test/java/org/takes/http/BkParallelTest.java]: I don't think it's the right suggestion. mocking is in general a bad practice: http://www.yegor256.com/2014/09/23/built-in-fake-objects.html

**Comment by yegor256** [comment]
@dmzaytsev one comment from me

**Comment by dmzaytsev** [comment]
@yegor256 I have changed the todo description 
looks good now ?

**Comment by yegor256** [comment]
@rultor merge

**Comment by m4solutions** [comment]
Close

**Comment by dmzaytsev** [comment]
@m4solutions you shoud close the issue instead the PR

**Comment by m4solutions** [comment]
@dmzaytsev thanks for pointing out, already closed

**Comment by davvd** [comment]
@elenavolokhova please, let us know what do you think about this ticket, according to our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by elenavolokhova** [comment]
@pinaf There were only two issues found during review. Both of them a rather cosmetic then design issues. Try to pay attention to bigger design problems and the overall quality of the code.

**Comment by pinaf** [comment]
@elenavolokhova sure - but puzzle length is actually enforced :) thanks!

**Comment by elenavolokhova** [comment]
@pinaf Yes, I know :)

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks a lot, next time everybody should try to make it better

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks for the review, we'll try better next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks a lot, next time everybody should try to make it better

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks for the review, we'll try better next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thank you, let's try to make it "good" next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
@pinaf I added 10 mins to @elenavolokhova (for QA review) in transaction 56876720; thanks, paid, **18 mins** to your account, payment ID is `AP-90M531492P526014X`, task took 18 hours and 18 mins to complete; you're getting extra minutes here (c=3); +18 added to your rating, at the moment it is: [+7711](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**